Kudu.Core/Deployment/StructuredTextDocument.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Kudu.Core.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Kudu.Core.Deployment
+{
+    public class StructuredTextDocument<T>
+    {
+        private readonly static object DocumentLock = new object();
```

Comment:
"DocumentLock" should per log file? shouldn`t be static? 

right now all logs are share the same lock object